sale_import_base/components/importer_sale_channel.py

```diff
@@ -0,0 +1,365 @@
+import base64
+import datetime
+import json
+from copy import deepcopy
+
+from marshmallow_objects import ValidationError as MarshmallowValidationError
+
+from odoo.exceptions import ValidationError
+
+from odoo.addons.component.core import Component
+
+MAPPINGS_SALE_ORDER_ADDRESS_SIMPLE = [
+    "name",
+    "street",
+    "street2",
+    "zip",
+    "city",
+    "email",
+]
+
+
+class ImporterSaleChannel(Component):
+    _inherit = "processor.import"
+    _name = "importer.sale.channel"
+    _apply_on = ["sale.order"]
+
+    def batch_run_chunks(self, imports):
+        """
+        :param imports: json dict of multiple chunks # DISCUSSION
+        :return: chunks
+        For an up-to-date version of the imports format, check datamodels
+        methods prefixed by "si" are related to sale imports
+        """
+        name = "SaleImport_" + str(datetime.datetime.today())
+        data_encoded = base64.b64encode(json.dumps(imports).encode("utf-8"))
+        attachment_vals = {"name": name, "db_datas": data_encoded, "store_fname": name}
+        attachment = self.env["ir.attachment"].create(attachment_vals)
+        result = self.env["queue.job.chunk"]
+        for el in imports:
+            dump = json.dumps(el)
+            chunk_vals = {
+                "usage": "import",
+                "apply_on_model": "sale.order",
+                "data_str": dump,
+                "model_id": "ir.attachment",
```

Comment:
we do not need ir.attachment, just link the chunk to the sale channel
model_name : "sale.channel"
record_id: channel_id